framework/src/modules/bft/module.ts

```diff
@@ -49,6 +60,18 @@ export class BFTModule extends BaseModule {
 	// eslint-disable-next-line @typescript-eslint/no-empty-function
 	public async afterGenesisBlockExecute(_context: GenesisBlockExecuteContext): Promise<void> {}
 
-	// eslint-disable-next-line @typescript-eslint/no-empty-function
-	public async afterBlockExecute(_context: BlockAfterExecuteContext): Promise<void> {}
+	public async afterBlockExecute(context: BlockAfterExecuteContext): Promise<void> {
+		const votesStore = context.getStore(this.id, STORE_PREFIX_BFT_VOTES);
+		const paramsStore = context.getStore(this.id, STORE_PREFIX_BFT_PARAMETERS);
+		const paramsCache = new BFTParametersCache(paramsStore);
+		const bftVotes = await votesStore.getWithSchema<BFTVotes>(EMPTY_KEY, bftVotesSchema);
+
+		insertBlockBFTInfo(bftVotes, context.header, this._maxLengthBlockBFTInfos);
+		await updatePrevotesPrecommits(bftVotes, paramsCache);
+		await updateMaxHeightPrevoted(bftVotes, paramsCache);
+		await updateMaxHeightPrecommitted(bftVotes, paramsCache);
+		updateMaxHeightCertified(bftVotes, context.header);
+		await votesStore.setWithSchema(EMPTY_KEY, bftVotes, bftVotesSchema);
+		await deleteBFTParameters(paramsStore, bftVotes.maxHeightCertified + 1);
```

Comment:
I am not sure this step is correct. In the LIP it says

> Let `h` be the largest integer such that `h <= bftVotes.maxHeightCertified+1` and there is an entry in the BFT Parameters store with key `h`. Then remove any key-value pair from BFT Parameters with a key strictly smaller than `h`.

Assume `bftVotes.maxHeightCertified == 400` and `h == 310` as there is no larger entry in the BFT Parameters store at height at most 400. Wouldn't the entry at height `h` be removed here?

Reply:
https://github.com/LiskHQ/lisk-sdk/pull/6796/files#diff-b0486fd5b86e4f912f1d74c0eca9757dfee97cbca56340a907839686961d99c7R41-R55

in that case, the input will be 401 and since height 401 doesn't exist, it will not delete anything with above code.
it will delete if height `401` exist in the params store,  and it will delete all the params from `0 - 400`

Reply:
> it will delete all the params from 0 - 400

But this would be incorrect, as the entry with key 310 needs to be maintained.

Reply:
> it will delete all the params from 0 - 400

will happen only if `if height 401 exist in the params store`, but that's not correct?

Reply:
Ah, I missed the following line above
```
if (!exist) {
    return;
}
```
I still think the logic is incorrect. Assume we have the following entries in the BFT Parameters store:
104 -> ... 
207 -> ...
310 -> ..
Then according to the LIP, all entries, except the largest one that is at most 401, should be removed. If I understand the implementation correctly, no entry would be removed as there is no entry for height 401.

Reply:
i think it's doing
there is an entry in the BFT Parameters store with key `401`. Then remove any key-value pair from BFT Parameters with a key strictly smaller than `401`.
so if there is no entry for height 401, it doesn't delete anything.
so if `h` is 310`, it will delete 104 and 207?

Reply:
or when it's called with `401` and there is not entry for 401, it should delete everything except `310`?

Reply:
> or when it's called with 401 and there is not entry for 401, it should delete everything except 310?

Yep, that is correct.

Reply:
I think we clarified this point so the comment can be resolved.